### PR TITLE
Added draft v1 of jaxb contract first

### PR DESCRIPTION
**Pull request by garethahealy**
This PR:
- Adds a JAXB based model, built originally from the schema.
- Model can also be used via a fluent API, i.e.: .withThingToChange(thing)
- This feature is not enabled by default, but can be via a property
- The reason for the elengine package model is due to not being able to insert a ELEngine instance when JAXB is parsing.

This feature wont be abled by default for a few releases, just to make sure nothing has been broken.

CC @orange-buffalo for code review.

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -249,19 +246,38 @@
                 <artifactId>jaxb2-maven-plugin</artifactId>
                 <executions>
                     <execution>
+                        <id>schemagen</id>
+                        <phase>generate-resources</phase>
+                        <goals>
+                            <goal>schemagen</goal>
```
Comment (orange-buffalo) [inline core/pom.xml]: I remember last time we had a discussion about using Lombok in model classes, to remove those boilerplate getters, setters and toStrings, but after adding Lombok the build was failing. I created a mojohaus/jaxb2-maven-plugin#88 to address this, but did not get any response yet. 

But, I was able to add Lombok and build the project with previous major version of the jaxb2 plugin (they rewrote a lot in 2x). In my opinion, clean code in core is more important that using the latest maven plugin version, thus I would recommend to apply this solution. 

In order to build with lombok, I changed the `jaxb2-maven-plugin`version to `1.6` and its execution configuration to the following:
```
                     <execution>
                        <id>schemagen</id>
                        <phase>generate-resources</phase>
                        <goals>
                            <goal>schemagen</goal>
                        </goals>
                        <configuration>
                            <workDirectory>${project.build.directory}/generated-resources/schemagen</workDirectory>
                            <includes>
                                org/dozer/builder/model/jaxb/**
                            </includes>
                            <transformSchemas>
                                <transformSchema>
                                    <uri>http://dozermapper.github.io/schema/bean-mapping</uri>
                                    <toFile>bean-mapping.xsd</toFile>
                                </transformSchema>
                            </transformSchemas>
                        </configuration>
                    </execution>
                    <execution>
                        <id>xjc</id>
                        <phase>generate-test-sources</phase>
                        <goals>
                            <goal>xjc</goal>
                        </goals>
                        <configuration>
                            <schemaDirectory>
                                ${project.basedir}/src/test/xsd/jaxb
                            </schemaDirectory>
                            <packageName>org.dozer.vo.jaxb.employee</packageName>
                            <outputDirectory>target/jaxb-sources</outputDirectory>
                        </configuration>
                    </execution>
``` 
 
After that the plugin generated the XSD from classes with Lombok annotations applied.

File: core/src/main/java/org/dozer/builder/model/jaxb/ClassDefinition.java [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+
+    public DozerClass build(BeanContainer beanContainer) {
+        DozerClass dozerClass = new DozerClass(beanContainer);
+        dozerClass.setName(clazz);
+        dozerClass.setBeanFactory(beanFactory);
+        dozerClass.setFactoryBeanId(factoryBeanId);
+        dozerClass.setMapGetMethod(mapGetMethod);
+        dozerClass.setMapSetMethod(mapSetMethod);
+        dozerClass.setCreateMethod(createMethod);
+        dozerClass.setMapNull(mapNull == null ? null : mapNull);
```
Comment (orange-buffalo) [inline core/src/main/java/org/dozer/builder/model/jaxb/ClassDefinition.java]: Maybe we could just pass the `mapNull` as the result of ternary operator will be always equal to the `mapNull`?

File: core/src/main/java/org/dozer/builder/xml/BeanMappingXMLBuilder.java [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+            MappingUtils.throwMappingException(ex);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<MappingFileData> build(BeanContainer beanContainer, DestBeanCreator destBeanCreator, PropertyDescriptorFactory propertyDescriptorFactory) {
+        List<MappingFileData> answer = new ArrayList<MappingFileData>();
```
Comment (orange-buffalo) [inline core/src/main/java/org/dozer/builder/xml/BeanMappingXMLBuilder.java]: We could cleanup a bit here and use  the diamond operator.

File: spring/src/test/java/com/github/dozermapper/spring/DozerBeanMapperFactoryBeanTest.java [hunk 4]
```diff
@@ -68,10 +68,10 @@ public void testOk() throws Exception {
         assertTrue(factory.isSingleton());
 
         DozerBeanMapper mapper = (DozerBeanMapper)factory.getObject();
-        List<?> files = mapper.getMappingFiles();
+        //List<?> files = mapper.getMappingFiles();
```
[the remark this answers is not in the document]
Comment (garethahealy) [reply spring/src/test/java/com/github/dozermapper/spring/DozerBeanMapperFactoryBeanTest.java]: Not sure why that was commented out. Have uncommented and test still works.

File: core/src/main/java/org/dozer/builder/xml/BeanMappingXMLBuilder.java (hunk 3 above)
Comment (garethahealy) [reply core/src/main/java/org/dozer/builder/xml/BeanMappingXMLBuilder.java]: done.

File: core/pom.xml (hunk 1 above)
Comment (garethahealy) [reply core/pom.xml]: Done, will start doing the lombok changes when i get some more time

**Comment by garethahealy** [comment]
@orange-buffalo ; any further comments regarding review?

**Pull request merged**